tests/test_unit.py

```diff
@@ -184,92 +165,82 @@ def test_reference_in_summary_more_xrefs(self):
         See <xref uid="google.cloud.requests.Session.request">request()</xref> documentation for details.
-"""
-        lines_want = lines_want.split("\n")
-        xrefs_want = [
-          "google.cloud.requests.Session",
-          "google.cloud.requests.tuple",
-          "google.cloud.requests.Session.request"
-        ]
-
-        self.assertEqual(lines_got, lines_want)
-        self.assertCountEqual(xrefs_got, xrefs_want)
-        # assertCountEqual is a misleading name but checks that two lists contain
-        # same items regardless of order, as long as items in list are sortable.
+            """,
+            [
+              "google.cloud.requests.Session",
+              "google.cloud.requests.tuple",
+              "google.cloud.requests.Session.request",
+            ],
+        ],
+    ]
+    @parameterized.expand(test_entries)
+    def test_resolves_references_in_summary(
+        self,
+        test_docstring,
+        expected_content,
+        expected_xrefs,
+    ):
+        xrefs_to_check = []
+        # Resolve over different regular expressions for different types of reference patterns.
+        content_to_resolve, xrefs = (
+            extension._resolve_reference_in_module_summary(
+                extension.REF_PATTERN,
+                test_docstring.split("\n"),
+            )
+        )
+        xrefs_to_check.extend(xrefs)
 
+        resolved_content, xrefs = (
+            extension._resolve_reference_in_module_summary(
+                extension.REF_PATTERN_LAST,
+                content_to_resolve,
+            )
+        )
+        xrefs_to_check.extend(xrefs)
 
-    # Variables used for testing _extract_docstring_info
-    top_summary1_want = "\nSimple test for docstring.\n\n"
-    summary_info1_want = {
-        'variables': {
-            'arg1': {
-                'var_type': 'int',
-                'description': 'simple description.'
-            },
-            'arg2': {
-                'var_type': 'str',
-                'description': 'simple description for `arg2`.'
-            }
-        },
-        'returns': [
-            {
-                'var_type': 'str', 
-                'description': 'simple description for return value.'
-            }
-        ],
-        'exceptions': [
-            {
-                'var_type': 'AttributeError', 
-                'description': 'if `condition x`.'
-            }
-        ]
-    }
+        self.assertEqual(resolved_content, expected_content.split("\n"))
+        self.assertCountEqual(xrefs_to_check, expected_xrefs)
 
 
-    # Test for resolving square bracketed references.
-    def test_reference_square_brackets(self):
-        xrefs_want = [
-            'google.cloud.kms.v1.KeyRing.name',
-            'google.cloud.kms.v1.KeyRing',
-            'google.cloud.kms.v1.ImportJob',
+    def test_resolves_square_bracket_references(self):
+        expected_xrefs = [
```

Comment:
Nit: I think this can also be a test parameter, just for consistency with the other tests your refactored!

Reply:
Agreed. Done.